drivers/video/video_mcux_mipi_csi2rx.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mipi_csi2rx
+
+#include <fsl_mipi_csi2rx.h>
+
+#include <zephyr/drivers/video.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mipi_csi);
+
+#define DEFAULT_MIPI_CSI_NUM_LANES 2
+#define DEFAULT_CAMERA_FRAME_RATE  30
```

Comment:
Should we make this configurable with devicetree?

Reply:
> There are a few "parameters" that can be declared in the devicetree, for example
>
>   -  the camera orientation / rotation
 >  -  data-lanes / clock-lanes

Isn't one of the parameters mentioned here to move to devicetree the number of data lanes? It seems like that is an excellent candidate for devicetree. As far as the framerate, I do agree this might not be best for devicetree. *Maybe* Kconfig? I'm not sure where best to put this.

Reply:
The _ov5640 driver does not support framerate changing_, (even the much more complete ov5640 driver in Linux only supports framerate changing in parallel mode, not csi). That's the main point.

Even it supports framerate changing. As being said, this needs to be done in a _negotiation at runtime_, between application, camera sensor, mipi csi2, csi (similar to set_format)... A simple expose in Kconfig or devicetree will not work, I think.

Reply:
Ok- thanks for the clarification :) I wasn't aware that there would need to be runtime negotiation here- we can leave this value as is